soc/xlnx/mbv32/soc.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 - 2024 Advanced Micro Devices, Inc.
+ * Copyright (c) 2023 Alp Sayin <alpsayin@gmail.com>
+ * Copyright (c) 2017 Jean-Paul Etienne <fractalclone@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <zephyr/arch/common/sys_io.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/arch/riscv/irq.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/interrupt_controller/intc_xlnx.h>
+#include <zephyr/init.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/tracing/tracing.h>
+
+/**
+ * @file
+ *
+ * The file is wiring architecture irq handler with secondary interrupt
+ * controller. AXI intc is used as secondary interrupt controller. Another
+ * level is not supported now but can be added in future. Maximum amount
+ * interrupts handled by AXI intc is 32.
+ */
+
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(soc);
+
+/**
+ * @brief Override arch_irq_enable with a call to intc driver
```

Comment:
If this is call to intc driver, why not provide `arch_irq_enable()` in the driver? Some driver does it under `drivers/interrupt_controller/`.

Reply:
Please correct me if I am wrong but arch_irq_enable() is used for primary interrupt controllers. In our case intc is secondary controller because primary is directly inside RISC-V core.

Reply:
Is your intc used by multiple SOCs or other arch, such as plain Microblaze? If so, it should be placed under driver, IMHO.